Library/Homebrew/cmd/uninstall.rb

```diff
@@ -9,15 +10,41 @@ def uninstall
       ARGV.kegs.each do |keg|
         keg.lock do
           puts "Uninstalling #{keg}... (#{keg.abv})"
+
+          # Find old name and old name symlinks for formula and remove them
+          begin
+            formula = Formulary.from_rack(keg.rack)
```

Comment:
> I need formula to get old name

You can get this information directly from files system. If `Cellar/foo` is a symbol link to `Cellar/bar`, then foo is an old name.